modules/artifact/src/main/java/org/mule/module/artifact/classloader/IBMMQResourceReleaser.java

```diff
@@ -396,36 +398,50 @@ private void processThreadLocalMap(Field threadLocalMapTableField, Object thread
    * Close JMSCCThreadPoolMaster threads pending to close to avoid thread leaks after redeployment
    */
   private void disposeMQThreads() {
-    List<Thread> threads = ThreadUtils.getAllThreads().stream()
-        .filter(thread -> thread.getName().equals(JMSCC_THREAD_POOL_MAIN_NAME))
-        .filter(thread -> thread.getClass().getClassLoader() == driverClassLoader)
+    String shouldAvoid = System.getProperty("avoid.dispose.mq.threads");
+    if (Boolean.valueOf(shouldAvoid)) {
+      return;
+    }
+
+    List<Thread> threads = getAllThreads().stream()
+        .filter(isThreadToBeDisposed())
         .collect(Collectors.toList());
 
     if (!threads.isEmpty()) {
-      threads.forEach(thread -> killThread(thread));
+      threads.forEach(thread -> closeWorkerThread(thread));
       try {
         killIBMWorker();
       } catch (Throwable e) {
-        LOGGER.debug("An error occurred trying to close the WorkQueueManager", e);
+        LOGGER.error("An error occurred trying to close the WorkQueueManager", e);
       }
     }
   }
 
   /**
-   * Kill any thread alive
+   * @return Predicate to evaluate if thread is disposable
+   */
+  private Predicate<Thread> isThreadToBeDisposed() {
+    Predicate<Thread> nameThreadPredicate = t -> t.getName().equals(JMSCC_THREAD_POOL_MAIN_NAME);
+    Predicate<Thread> classLoaderPredicate = t -> t.getClass().getClassLoader() == driverClassLoader;
+
+    return nameThreadPredicate.and(classLoaderPredicate);
+  }
```

Comment:
Sorry, I didn't mean to have this function return a Predicate, that is too convoluted.
I meant the function to return a boolean and use it as a Predicate via method reference:
```suggestion
  private boolean isThreadToBeDisposed(Thread thread) {
    return thread.getName().equals(JMSCC_THREAD_POOL_MAIN_NAME) &&
        thread.getClass().getClassLoader() == driverClassLoader;
  }
```
(Update the JDoc too)